### PR TITLE
Add more infos to the README

### PR DESCRIPTION
**Pull request by xtuc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,24 +2,56 @@
...
 
+Optionally the developers can specify an SRI hash which semantically works like an assertions:
+
+```js
+import json from "./foo.json" with
+    type: "json",
+    integrity: "sha384-some-hash";
+```
+
+#### Builtin modules
```
Comment (littledan) [inline README.md]: I don't really think we should include built-in modules here. Let's let that proceed separately first.

File: README.md [hunk 2]
```diff
@@ -2,24 +2,56 @@
...
 
+The ImportDeclaration would allow any arbitrary attributes after the `with` keyword.
+
+A `type` key should be mandatory to load a module that isn't JavaScript (json, css, html, ...).
+
 ```js
 import json from "./foo.json" with type: "json";
 ```
 
+Optionally the developers can specify an SRI hash which semantically works like an assertions:
```
Comment (littledan) [inline README.md]: Let's be clear that this is a separate possible proposal; at first, we'd just do `type:`.

File: README.md [hunk 3]
```diff
@@ -2,24 +2,56 @@
...
-### When would the checks happen
+"The author is the only true arbiter of the parse goal of content".
+
+### Semantics
+
+On the web, before any code executes this semantics apply before any code executes.
+
+#### `type`
+
+Check that the mimetype matches what type was specified in the attributes.
```
Comment (littledan) [inline README.md]: Note, the type in the attributes isn't a mimetype itself, but rather implies a mimetype.

File: README.md [hunk 4]
```diff
@@ -30,16 +62,24 @@ import("foo.json", { type: "json" })
 <script src="foo.wasm" type="webassembly"></script>
 ```
 
+While the browser can check for WebAssembly's magic number. We would like to offer a way for the developer to assert the type of their ressources.
```
Comment (littledan) [inline README.md]: Namely, WebAssembly validation will fail if this magic number is not correct. We'd assert that the mimetype matches the type before even beginning to run WebAssembly validation.

File: README.md [hunk 5]
```diff
@@ -30,16 +62,24 @@ import("foo.json", { type: "json" })
...
+
 ### WebAssembly
 
-Custom section: per import specifier, what options are needed
+Introduce a new custom section (named `importattributes`) that will annotate with attributes each imported module (based on the import section).
+
+Web WebAssembly embedding should check the attributes.
```
Comment (littledan) [inline README.md]: This phrasing is a bit turned around. I'd imagine these attributes would be passed to HTML, which would be the one who does the checking.

Comment (xtuc) [reply README.md]: Removed this line, it might be clear already.

**Pull request merged**